core/src/main/java/org/radargun/config/ClasspathScanner.java

```diff
@@ -22,94 +19,57 @@
+                                                                             String requirePackage,
+                                                                             Consumer<Class<? extends TClass>> consumer) {
+      
+      FastClasspathScanner fcs;
+      if (requirePackage != null) {
+         fcs = new FastClasspathScanner(requirePackage);
+      } else {
+         fcs = new FastClasspathScanner();
       }
-   }
-
-   public static <TClass, TAnnotation extends Annotation> void scanJar(String path,
-                                                                       Class<TClass> superClass, Class<TAnnotation> annotationClass, String requirePackage, Consumer<Class<? extends TClass>> consumer) {
-      log.tracef("Looking for @%s %s, loading classes from %s", annotationClass.getSimpleName(), superClass.getSimpleName(), path);
-      try (ZipInputStream inputStream = new ZipInputStream(new FileInputStream(path))) {
-         for (; ; ) {
-            ZipEntry entry = inputStream.getNextEntry();
-            if (entry == null) break;
-            if (!entry.getName().endsWith(CLASS_SUFFIX)) continue;
-            String className = entry.getName().replace('/', '.').substring(0, entry.getName().length() - CLASS_SUFFIX.length());
-            scanClass(className, superClass, annotationClass, requirePackage, consumer);
-         }
-      } catch (FileNotFoundException e) {
-         log.error("Cannot load executed JAR file '" + path + "'to find stages.");
-      } catch (IOException e) {
-         log.error("Cannot open/read JAR '" + path + "'");
-      }
-   }
-
-   protected static <TClass, TAnnotation extends Annotation> void scanClass(String className,
-                                                                            Class<TClass> superClass, Class<TAnnotation> annotationClass, String requirePackage, Consumer<Class<? extends TClass>> consumer) {
-      if (requirePackage != null && !className.startsWith(requirePackage)) return;
-      Class<?> clazz;
-      try {
-         System.setErr(NULL_PRINT_STREAM); // suppress any error output
-         clazz = Class.forName(className);
-         System.setErr(ERR_PRINT_STREAM);
-      } catch (ClassNotFoundException e) {
-         log.trace("Cannot load class " + className, e);
-         return;
-      } catch (NoClassDefFoundError e) {
-         log.trace("Cannot load class " + className, e);
-         return;
-      } catch (LinkageError e) {
-         log.trace("Cannot load class " + className, e);
-         return;
-      } catch (Throwable e) {
-         // static ctor can throw non-wrapped error
-         log.trace("Cannot load class " + className, e);
-         return;
-      }
-      TAnnotation annotation = clazz.getAnnotation(annotationClass);
-      if (annotation != null) {
-         if (!superClass.isAssignableFrom(clazz)) {
+      
+      List<String> matches = fcs.scan()
+            .getNamesOfClassesWithAnnotation(annotationClass.getName());
```

Comment:
Could you include the check for superclass as well into filtering (so that you don't classload the file before you're sure you want it)?